typescript/infra/src/config/agent.ts

```diff
@@ -304,22 +306,23 @@ export class ChainAgentConfig<Chain extends ChainName> {
         let validator: KeyConfig = {
```

Comment:
in the line above, I think we should filter validators out that are readonly before the `map`. The intent of this function is to return an array of helm validator configs for validators that we want to run in K8s - meaning if there are any readonly validators configured, as it is now we'll end up trying to run that validator in K8s. If we filter then we can remove the `if (!val.readonly) {` you added

(probably worth a comment in here indicating this is the intent of the function because it's pretty unclear to readers)